src/main/test/surfrdUtils_test/test_surfrdUtils.pf

```diff
@@ -43,6 +43,114 @@ contains
 
   end subroutine setup_test_collapse_routines
 
+  @Test
+  subroutine test_collapse_individual_lunits()
+     ! This test sets the user-defined thresholds toosmall_*
+     ! (soil, crop, glacier, lake, wetland, urb_tbd, urb_hd, urb_md)
+     ! to a range of values and the subroutine call returns and checks the
+     ! updated wt_lunit values.
+     ! Tests performed:
+     ! - toosmall_* = 0 ... DO NOTHING
+     ! - toosmall_* = 1 ... Remove landunits with <= 1% of the gridcell area
+     ! - toosmall_* = 8 ... Remove landunits with <= 8% of the gridcell area
+     ! - toosmall_* = 39 ... Remove landunits with <= 39% of the gridcell area
+     ! - toosmall_* = 100 ... Keep only the dominant landunit per gridcell
+     use pftconMod, only: pftcon
+     use clm_instur, only: wt_lunit
+     use landunit_varcon, only: max_lunit, istsoil, istcrop, istice_mec, &
+                                istdlak, istwet, isturb_tbd, isturb_hd, &
+                                isturb_md
+
+     implicit none
+     integer :: test  ! test number in loop of tests
+     integer, parameter :: begg = 2, endg = 4, tests = 5
+     integer, parameter :: toosmall_soil(tests) = (/ 0, 1, 8, 39, 100/)
+     integer, parameter :: toosmall_crop(tests) = (/ 0, 1, 8, 39, 100/)
+     integer, parameter :: toosmall_glacier(tests) = (/ 0, 1, 8, 39, 100/)
+     integer, parameter :: toosmall_lake(tests) = (/ 0, 1, 8, 39, 100/)
+     integer, parameter :: toosmall_wetland(tests) = (/ 0, 1, 8, 39, 100/)
+     integer, parameter :: toosmall_urb_tbd(tests) = (/ 0, 1, 8, 39, 100/)
+     integer, parameter :: toosmall_urb_hd(tests) = (/ 0, 1, 8, 39, 100/)
+     integer, parameter :: toosmall_urb_md(tests) = (/ 0, 1, 8, 39, 100/)
+     real(r8), allocatable :: wt_expected(:,:)
+     real(r8), allocatable :: wt_in_out(:,:)  ! used in subr. call
+     real(r8) :: expctd(9)
+     real(r8), parameter :: input_value(8) = (/40._r8, 33._r8, 12._r8, 7._r8, 5._r8, 2._r8, 1._r8, 0._r8/)
+
+     ! Set relevant pftcon values to defaults; override where necessary
+     call pftcon%InitForTesting()
+     allocate( wt_lunit(begg:endg,istsoil:max_lunit) )
+     allocate( wt_expected(begg:endg,istsoil:max_lunit) )
+     allocate( wt_in_out(begg:endg,istsoil:max_lunit) )
+
+     ! INPUT VALUES
+     wt_lunit(begg:,:) = 0._r8  ! initialize
+     wt_lunit(begg:,istsoil) = (/ input_value(1), input_value(8), input_value(4)/)
```

Comment:
It's hard to make sense of the array indices here. I don't know why index 1, 8, and 4 are chosen here.

Reply:
Added comment in the code to explain.